extensions/amp-app-banner/1.0/component/android.js

```diff
@@ -0,0 +1,104 @@
+import {docInfoService} from '#preact/services/document';
+import {platformService} from '#preact/services/platform';
+import {timerService} from '#preact/services/timer';
+import {urlService} from '#preact/services/url';
+import {viewerService} from '#preact/services/viewer';
+import {xhrService} from '#preact/services/xhr';
+
+import {user} from '#utils/log';
+
+import {openWindowDialog} from '../../../../src/open-window-dialog';
+
+const OPEN_LINK_TIMEOUT = 1500;
+
+/**
+ * @return {{openOrInstall: function(): void, promise: Promise<Response>}|null}
+ */
+export function getAndroidAppInfo() {
+  const win = self.window;
+  // We want to fallback to browser builtin mechanism when possible.
+  const isChromeAndroid =
+    platformService.isAndroid() && platformService.isChrome();
+  const isProxyOrigin = urlService.isProxyOrigin(win.location);
```

Comment:
This feels icky. Proxy origin and source origin are relevant only to the AMP format.

Reply:
Removed.